inc/shortcodes/class-script.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Shortcake_Bakery\Shortcodes;
+
+class Script extends Shortcode {
+
+	private static $whitelisted_script_domains = array();
+
+	/**
+	 * Get the whitelisted script domains for the plugin
+	 */
+	public static function get_whitelisted_script_domains() {
+		return apply_filters( 'shortcake_bakery_whitelisted_script_domains', self::$whitelisted_script_domains );
+	}
+
+	public static function reversal( $content ) {
+
+		$whitelisted_script_domains = self::get_whitelisted_script_domains();
+		$content = preg_replace_callback( '!\<script\s[^>]*?src=[\"\']([^\"\']+)[\"\'][^>]*?\>\s{0,}\</script\>!i', function( $match ) use ( $whitelisted_script_domains ) {
+
+			$url = ( 0 === strpos( $match[1], '//' ) ) ? 'http:' . $match[1] : $match[1];
+			$host = parse_url( $url, PHP_URL_HOST );
+			if ( ! in_array( $host, $whitelisted_script_domains ) ) {
+				return $match[0];
+			}
+			$shortcode_tag = self::get_shortcode_tag();
```

Comment:
Yes, it looks like other shortcodes use `preg_match_all` and then loop through the results, so that they stay in the same context, rather than `preg_match_callback` with a closure creating a new context.